src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectOptionsProcessor.cs

```diff
@@ -236,12 +245,26 @@ private void RuleSetFile_UpdatedOnDisk(object? sender, EventArgs e)
             // effective values was potentially done by the act of parsing the command line. Even though the command line didn't change textually,
             // the effective result did. Then we call UpdateProjectOptions_NoLock to reapply any values; that will also re-acquire the new ruleset
             // includes in the IDE so we can be watching for changes again.
-            var commandLine = _commandLineStorage == null ? ImmutableArray<string>.Empty : _commandLineStorage.ReadLines();
+            var commandLine = _commandLineStorageHandle == null
+                ? ImmutableArray<string>.Empty
+                : EnumerateLines(_temporaryStorageService, _commandLineStorageHandle.Identifier).ToImmutableArray();
 
             DisposeOfRuleSetFile_NoLock();
             ReparseCommandLine_NoLock(commandLine);
             UpdateProjectOptions_NoLock();
         }
+
+        static IEnumerable<string> EnumerateLines(
+            ITemporaryStorageServiceInternal temporaryStorageService,
+            TemporaryStorageIdentifier storageIdentifier)
+        {
+            using var stream = temporaryStorageService.ReadFromTemporaryStorageService(storageIdentifier, CancellationToken.None);
+            using var reader = new StreamReader(stream);
+
+            string? line;
+            while ((line = reader.ReadLine()) != null)
```

Comment:
>while ((line = reader.ReadLine()) != null) [](http://example.com/codeflow?start=12&length=42)

nit:

while (reader.ReadLine() is string line)

Reply:
done.